### ISSUE TITLE
document

### ISSUE THREAD

**Issue by timelyportfolio** [timelyportfolio/dataui#2]
I believe the API is stable enough to begin documenting in `roxygen` and  also in other supplementary materials, such as vignettes and articles.

**Comment by JasonCurole** [comment]
@timelyportfolio are you still looking for support on this?

**Comment by timelyportfolio** [comment]
@JasonCurole yes definitely - documentation is the only obstacle between current state and CRAN.  Happy to discuss.

**Comment by JasonCurole** [comment]
Great. Happy to help. I'm new to contributing, but I've been using dataui (both on its own and through reactablefmtr) and am very appreciative. Let me know the best way for us to discuss.

**Comment by timelyportfolio** [comment]
@JasonCurole thanks so much again for potentially helping.  I forgot that I started documentation very similar to what I envision for the remaining componentss in [`dui_histogram`](https://github.com/timelyportfolio/dataui/blob/master/R/duihistogram.R#L1-L46) and [`dui_sparkline`](https://github.com/timelyportfolio/dataui/blob/master/R/duisparkline.R#L1-L51).  Would this be sufficient to establish a pattern that we might be able to use as a model for the components in [components.R](https://github.com/timelyportfolio/dataui/blob/master/R/components.R)?  Most of the arguments are in [storybook | props table tab](https://williaster.github.io/data-ui/?selectedKind=sparkline&selectedStory=Kitchen%20sink&full=0&addons=0&stories=1&panelRight=0) and/or in [github repo readme](https://github.com/williaster/data-ui/tree/master/packages/sparkline#series).  I remember not all of these are exactly correct or maybe even available, but I did not list these back when I was working on building the package.

Also, I think instead of `...` as the arguments for components, an R user might benefit from named arguments similar to [`dui_sparkline` arguments](https://github.com/timelyportfolio/dataui/blob/master/R/duisparkline.R#L53-L67).

I started with `dui_sparkbarseries` in this [commit](https://github.com/timelyportfolio/dataui/compare/master...documentation#diff-711086ce1e2e666b90a7039bbe9ba44d1b76eeb972a6b85f8dc21b7f54d42794) as an example.

If you still have interest, document away :).  Let me know how I can help.
